### ISSUE TITLE
`duplicate-names-in-diff-context` lint wrongly flags disjoint sibling branches with same-named objects

### ISSUE THREAD

**Issue by volodya-lombrozo** [objectionary/lints#972]
**What happens:** The `duplicate-names-in-diff-context` lint flags any object name that appears more than once anywhere in a file, regardless of nesting depth or whether the two objects could ever actually collide. It groups every `o` node by `@name` for the whole file in one flat pass, with no check for whether one node is actually an ancestor of the other (true shadowing) versus simply sitting in unrelated branches of the tree that are reached through different qualified paths and can never be confused.

Example (`fs/path.eo` in `eo-runtime`):
```eo
[uri] > posix
  ...
  [] > basename
    ...
[uri] > win32
  ...
  [] > basename
    ...
```
`posix` and `win32` are siblings under `path`. Their `basename` (and `extname`, `dirname`, `normalized`, `resolved`, `is-absolute`) are always reached as `posix.basename` vs `win32.basename` -- there is no ambiguity, yet the lint warns on every pair.

Other occurrences with the same shape in `eo-runtime`:
- `i16.eo` / `i32.eo` / `i64.eo` -- sibling arithmetic methods reuse local names like `left` / `right` / `bts` / `value` in unrelated scopes.
- `io/console.eo:79,91,129,141` -- sibling decorator stages each independently define their own `read` / `write` helper.
- `ss/map.eo:42-47` vs `:131-134` -- `hash` / `key` / `value` / `prev` reused across unrelated sibling sub-objects.
- `nk/socket.eo` -- lifecycle-stage objects (`scoped-socket`, `connect`, `listen`, `accept`, `closed-socket`, `safe-socket`) each redeclare `sd` / `addr` / `sock` / etc. in their own disjoint scope -- ~20 warnings in one file.

**Comment by volodya-lombrozo** [comment]
I tried to fix the issue, but there are too many cases when we reuse the object names deliberately. It seems simpler to remove this check.

**Comment by volodya-lombrozo** [comment]
Moreover, we might have some sort of recursive code that also reuses names:

```
  [size] > read                   
    ((input-block bts --).read size).@ > @
    [data buffer] > input-block
      [size] > read                
        ...
        input-block ...            
```

(we can't just rename `read` attributes)

**Comment by yegor256** [comment]
@volodya-lombrozo ok, let's remove this lint

**Comment by volodya-lombrozo** [comment]
@rultor release, tag is `0.2.8`, title is `0.2.8`

**Comment by 0crat** [comment]
@volodya-lombrozo Thanks for closing this issue! You've earned +12 points for this: +8 as a basis; +4 for resolving it in less than 12 hours. Your running score is [+488](https://www.eolang.org/zerocracy/objectionary-vitals.html); don't forget to check your [Zerocracy account](https://www.zerocracy.com/) too).

**Comment by 0crat** [comment]
@volodya-lombrozo Thanks for reporting a new bug! You've earned +12 points. By reporting bugs, you help our project improve its quality. If you find anything else in the repository that [doesn't look](https://www.yegor256.com/2018/02/06/where-to-find-more-bugs.html) as good as you might expect, [do not hesitate](https://www.yegor256.com/2014/04/13/bugs-are-welcome.html) to [report](https://www.yegor256.com/2018/04/24/right-way-to-report-bugs.html) it. Your running score is [+500](https://www.eolang.org/zerocracy/objectionary-vitals.html); don't forget to check your [Zerocracy account](https://www.zerocracy.com/) too).
